tests/test_short_url.py

```diff
@@ -0,0 +1,84 @@
+"""Shortening url tests."""
+
+from urllib.parse import urlparse
+
+import pytest
+from pydata_sphinx_theme.short_link import ShortenLinkTransform
+
+
+class Mock:
+    """mock object."""
+
+    pass
+
+
+@pytest.mark.parametrize(
+    "platform,url,expected",
+    [
+        ("gitlab", "https://gitlab.com/tezos/tezos/-/issues", "tezos/tezos/issues"),
+        ("gitlab", "https://gitlab.com/tezos/tezos/issues", "tezos/tezos/issues"),
+        (
+            "gitlab",
+            "https://gitlab.com/gitlab-org/gitlab/-/issues/375583",
+            "gitlab-org/gitlab#375583",
+        ),
+        (
+            # todo, non cannonical url, needs extra parsing.
```

Comment:
is this TODO still outstanding? Looks like all tests are passing...

Reply:
I think that both gitlab.com/foo/bar/issues and gitlab.com/foo/bar/-/issues should shorten to the same.

It's more a good to have but would be an extra feature for another time.